frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/vm-settings-tab-state-update.ts

```diff
@@ -164,6 +171,101 @@ const osUpdater = ({ id, prevState, dispatch, getState }: UpdateOptions) => {
   }
 };
 
+const baseImageUpdater = ({ id, prevState, dispatch, getState }: UpdateOptions) => {
+  const state = getState();
+  if (
+    !hasVMSettingsValueChanged(
+      prevState,
+      state,
+      id,
+      VMSettingsField.OPERATING_SYSTEM,
+      VMSettingsField.FLAVOR,
+      VMSettingsField.WORKLOAD_PROFILE,
+    )
+  ) {
+    return;
+  }
+  if (iGetLoadedCommonData(state, id, VMWizardProps.isProviderImport)) {
+    return;
+  }
+
+  const relevantOptions = iGetRelevantTemplateSelectors(state, id);
+  const userTemplate = iGetVmSettingValue(state, id, VMSettingsField.USER_TEMPLATE);
+  const iCommonTemplates = iGetLoadedCommonData(state, id, VMWizardProps.commonTemplates);
+  const iTemplate =
+    iCommonTemplates && iGetRelevantTemplate(null, iCommonTemplates, relevantOptions);
```

Comment:
we don't have to lookup the iTemplate from iGetRelevantTemplate if we know userTemplate is selected

Reply:
done